parquet-column/src/main/java/org/apache/parquet/column/values/deltastrings/DeltaByteArrayWriter.java

```diff
@@ -82,11 +82,11 @@ public String memUsageString(String prefix) {
   @Override
   public void writeBytes(Binary v) {
     int i = 0;
-    byte[] vb = v.getBytes();
+    byte[] vb = v.getBytesUnsafe();
     int length = previous.length < vb.length ? previous.length : vb.length;
     for(i = 0; (i < length) && (previous[i] == vb[i]); i++);
     prefixLengthWriter.writeInteger(i);
-    suffixWriter.writeBytes(Binary.fromByteArray(vb, i, vb.length - i));
+    suffixWriter.writeBytes(Binary.fromUnmodifiedByteArray(vb, i, vb.length - i));
```

Comment:
I thought the contract for unsafe access to the bytes backing another Binary is to never update it, as it is owned by some other object. If that is true, then it should be up to the consumer what it wants to do with it as long as it never updates it.
On the same note, I have used fromUnmodifiedByteArray here as the producer is clearly not modifying the backing bytes.
However, I do understand your concern that this particular scenario acts as a producer of binary backed by bytes owned by some other entity and still claims that bytes won't be changed.
Two possible solutions, either change getBytesUnsafe() to getBytes(), or change fromUnmodifiedByteArray to inherit reusable property from original v, as suggested by you. Former will require an extra copy, which I think is not required here. I will update the code as suggested by you to inherit the reusable property from original v. Thanks for pointing this out.